stratified-sampling/script.whizzml

```diff
@@ -0,0 +1,41 @@
+;; Takes a dataset id and a catagorical field id, and breaks the
+;; dataset into pieces, one per value. Returns the list of datasets.
+
+(define (split-it ds-id fld-id)
+  (let (bucket-list (map (lambda (x) (head x)) 
+                         ((fetch ds-id) 
+                             ["fields" fld-id "summary" "categories"])))
+    (iterate (ds-list []            
+              bucket bucket-list)
+      (let (split-filter (flatline "(= (f {{fld-id}}) {{bucket}})")
+            split-ds (create-dataset {"origin_dataset" ds-id
```

Comment:
nit: this can also be written a bit more concisely as `(create-dataset ds-id {"lisp_filter" split-filter})`